### PR TITLE
#141 Store JSON response on /sdcard

### PR DESCRIPTION
**Pull request by jo-elimuai**
- Store AppCollection JSON so that the Custom Launcher can fetch the
AppCategories and AppGroups dynamically.

### CODE REVIEW

File: app/src/main/java/ai/elimu/appstore/synchronization/AppSynchronizationActivity.java [hunk 1]
```diff
@@ -248,17 +251,59 @@ private void processAppListData(@NonNull Response<ResponseBody> response) {
...
+                                        Timber.i("fileDir: " + fileDir);
+                                        if (!fileDir.exists()) {
+                                            fileDir.mkdirs();
+                                        }
+
+                                        File jsonFile = new File(fileDir, "app-collection.json");
+                                        Timber.i("jsonFile: " + jsonFile);
+                                        try {
+                                            FileUtils.writeStringToFile(jsonFile, jsonReponse, "UTF-8", true);
+                                        } catch (IOException e) {
```
Comment (tuancoltech) [inline app/src/main/java/ai/elimu/appstore/synchronization/AppSynchronizationActivity.java]: Redundant try catch here. It's already caught outside.

File: app/src/main/java/ai/elimu/appstore/synchronization/AppSynchronizationActivity.java [hunk 2]
```diff
@@ -248,17 +251,59 @@ private void processAppListData(@NonNull Response<ResponseBody> response) {
...
+                                        File fileDir = new File(Environment.getExternalStorageDirectory() + "/.elimu-ai/appstore/");
+                                        Timber.i("fileDir: " + fileDir);
+                                        if (!fileDir.exists()) {
+                                            fileDir.mkdirs();
+                                        }
+
+                                        File jsonFile = new File(fileDir, "app-collection.json");
+                                        Timber.i("jsonFile: " + jsonFile);
+                                        try {
+                                            FileUtils.writeStringToFile(jsonFile, jsonReponse, "UTF-8", true);
```
Comment (tuancoltech) [inline app/src/main/java/ai/elimu/appstore/synchronization/AppSynchronizationActivity.java]: You are appending json string to the file.
What if the app collection json is saved for the first time, then next time you launch the app, the collection content is updated?
The new json string will be appended to the existing file and it gives a wrong json.

File: app/src/main/java/ai/elimu/appstore/synchronization/AppSynchronizationActivity.java (hunk 1 above)
Comment (jo-elimuai) [reply app/src/main/java/ai/elimu/appstore/synchronization/AppSynchronizationActivity.java]: Fixed. Thanks :-)

File: app/src/main/java/ai/elimu/appstore/synchronization/AppSynchronizationActivity.java (hunk 2 above)
Comment (jo-elimuai) [reply app/src/main/java/ai/elimu/appstore/synchronization/AppSynchronizationActivity.java]: Ah, good observation, thanks! This has now been fixed.

**Pull request merged**